src/services/transactions_proposal.rs

```diff
@@ -31,3 +33,31 @@ pub fn submit_confirmation(
         ))
     }
 }
+
+pub fn propose_transaction(
+    context: &Context,
+    safe_address: &str,
+    transaction_request: &MultisigTransactionRequest,
+) -> ApiResult<()> {
+    let url = format!(
+        "{}/v1/safes/{}/multisig-transactions/",
+        base_transaction_service_url(),
+        &safe_address
+    );
+    let response = context
+        .client()
+        .post(&url)
+        .json(&transaction_request)
+        .send()?;
+
+    if response.status().is_success() {
+        let pattern = format!("*{}_{}*", &ID_PREFIX_MULTISIG_TX, &safe_address);
+        context.cache().invalidate_pattern(&pattern);
```

Comment:
This is a little dodgy, but also not really :/ Maybe it would be better to request the `transaction_id` instead, that way we know which details to invalidate in the cache? Although if this is a whole purpose endpoint, there might not be a `safe_tx_hash` to use for invalidation.

Reply:
Sorry that was poorly worded. I think that having a `tx_details_id` would make this invalidation pattern more accurate. But if this endpoint is meant to be used outside the `tx_details` eventually, then there might not be a `tx_details_id` to send at all.

Reply:
Why not use the same pattern as for the `submit_confirmation`? the `contract_transaction_hash` is the `safe_tx_hash`

Reply:
actually it might make sense to invalidate just by `safe_address`, as we want the list to be invalidated too (as we added a new tx)